config/config.go

```diff
@@ -59,10 +63,9 @@ func FromYAMLFile(name string, v Validator) error {
 // ParseFlags prints the help message to [os.Stdout] and exits.
 // Note that errors are not printed automatically,
 // so error handling is the sole responsibility of the caller.
-func ParseFlags(v any) error {
-	rv := reflect.ValueOf(v)
-	if rv.Kind() != reflect.Pointer || rv.IsNil() {
-		return errors.Wrapf(ErrInvalidArgument, "non-nil pointer expected, got %T", v)
+func ParseFlags[T any](v *T) error {
+	if v == nil {
+		return errors.Wrap(ErrInvalidArgument, "got nil pointer")
```

Comment:
I mean, which of our functions which take pointers checks them like this?

Reply:
https://github.com/Icinga/icinga-go-library/pull/72/files#r1768083795